# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/ContentParent.cpp
# Commit: 984db15bc9e2
# Full Hash: 984db15bc9e2b5baa20bdc86baaa1165c381d326
# Author: Kagami Sascha Rosylight <saschanaz@outlook.com>
# Date: 2024-11-08 04:19:40
# Regressor Bug: 1928702
# File Overlap Count: 1
# Description:
#   Bug 1928702 - Part 11: Migrate ShowPersistentNotification to use IPC r=asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D227696
# ==============================================================================

diff -r 622a66cdac2e -r 984db15bc9e2 dom/ipc/ContentParent.cpp
--- a/dom/ipc/ContentParent.cpp	Thu Nov 07 17:32:07 2024 +0000
+++ b/dom/ipc/ContentParent.cpp	Thu Nov 07 17:32:07 2024 +0000
@@ -4742,25 +4742,6 @@
   return IPC_OK();
 }
 
-mozilla::ipc::IPCResult ContentParent::RecvNotificationEvent(
-    const nsAString& aType, const NotificationEventData& aData) {
-  nsCOMPtr<nsIServiceWorkerManager> swm =
-      mozilla::components::ServiceWorkerManager::Service();
-  if (NS_WARN_IF(!swm)) {
-    // Probably shouldn't happen, but no need to crash the child process.
-    return IPC_OK();
-  }
-
-  MOZ_ASSERT(aType.EqualsLiteral("close"));
-  nsresult rv = swm->SendNotificationCloseEvent(
-      aData.originSuffix(), aData.scope(), aData.ID(), aData.title(),
-      aData.dir(), aData.lang(), aData.body(), aData.tag(), aData.icon(),
-      aData.data(), aData.behavior());
-  Unused << NS_WARN_IF(NS_FAILED(rv));
-
-  return IPC_OK();
-}
-
 mozilla::ipc::IPCResult ContentParent::RecvSyncMessage(
     const nsAString& aMsg, const ClonedMessageData& aData,
     nsTArray<StructuredCloneData>* aRetvals) {